src/pages/index.js

```diff
@@ -1,84 +1,188 @@
+import ConfirmationOnClick from "../components/scripts/popupWithConfirmation";
 import "../pages/index.css";
 
+// Api
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "d081ca18-dd96-4ff1-863a-c195fbcda74a",
+    "Content-Type": "application/json",
+  },
+});
+
+const userInformation = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__pic"
+);
+
+const formValidators = {};
+const enableValidation = (validationSettings) => {
+  DOM.forms.forEach((formElement) => {
+    const validator = new FormValidation(validationSettings, formElement);
+    const formName = formElement.getAttribute("name");
+    formValidators[formName] = validator;
+    validator.enableValidation();
+  });
+};
+
+let section = 0;
+
+Promise.all([api.getInitialCards(), api.getUserInfo()])
+  .then(([cardData, formData]) => {
+    section = new Section(
+      {
+        items: cardData,
+        renderer: (item) => {
+          const cardElement = createCard(item);
+          section.addItem(cardElement);
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
+
+    userInformation.setUserInfo(formData);
+    userInformation.setAvatar(formData.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+function handleDeleteCardClick(item) {
+  popupWithConfirmation.open();
+  popupWithConfirmation.setSubmitCall(() => {
+    popupWithConfirmation.deleting(true);
+    api
+      .deleteCard(item.getId())
+      .then(() => {
+        item.handleDeleteCard();
+        popupWithConfirmation.deleting(false);
+      })
+      .catch((err) => {
+        console.log("Error:", err);
+      });
+    popupWithConfirmation.close();
+  });
+}
+
+function handleImageLike(item) {
+  const newLikeStatus = !item.isLiked;
+  if (newLikeStatus) {
+    api
+      .likeCard(item.getId())
+      .then((respond) => {
+        item.setLikeStatus(respond.isLiked);
+      })
+      .catch((err) => {
+        console.log("Error:", err);
+      });
+  } else {
+    api
+      .unlikeCard(item.getId())
+      .then((respond) => {
+        item.setLikeStatus(respond.isLiked);
+      })
+      .catch((err) => {
+        console.error("Error:", err);
+      });
+  }
+}
+
+const popupWithConfirmation = new ConfirmationOnClick({
+  popupSelector: "#modal-delete-confirmation",
+});
+popupWithConfirmation.setEventListeners();
+
 // popup with form
 const newCardPopup = new PopupWithForm(
   "#profile-add-modal",
-  handleCardFormSubmit
+  handleCardAddFormSubmit
 );
 newCardPopup.setEventListeners();
+
 // edit popup form
-const userInfo = new UserInfo(".profile__title", ".profile__description");
-const popupEditForm = new PopupWithForm("#profile-edit-modal", (formData) => {
-  userInfo.setUserInfo(formData);
-  popupEditForm.close();
-});
+
+const popupEditForm = new PopupWithForm(
+  "#profile-edit-modal",
+  ({ name, about }) => {
+    popupEditForm.saving(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```